ProblemWidget.js

```diff
@@ -0,0 +1,53 @@
+define(function (require, exports, module) {
+    "use strict";
+
+    var InlineWidget = brackets.getModule("editor/InlineWidget").InlineWidget;
+
+    var widgetTemplate = require("text!templates/inlineWidgetLint.html");
+
+    function ProblemWidget(mark, line) {
+        InlineWidget.call(this);
+
+        var problems = [].concat(mark.warnings, mark.errors);
+
+        var $html = $(Mustache.render(widgetTemplate, { messages: problems }));
+
+        this.$wrapperDiv = $html;
+        this.$htmlContent.append(this.$wrapperDiv);
+
+        this.$wrapperDiv.on("mousedown", function (e) {
+            e.stopPropagation();
+        });
+
+        this.line = line;
+
+        this._sizeEditorToContent   = this._sizeEditorToContent.bind(this);
+    }
+
+    ProblemWidget.prototype = Object.create(InlineWidget.prototype);
+    ProblemWidget.prototype.constructor = ProblemWidget;
+    ProblemWidget.prototype.parentClass = InlineWidget.prototype;
+
+    ProblemWidget.prototype.$wrapperDiv = null;
```

Comment:
@Mark-Simulacrum I am going to merge this in.  Do we need `$wrapperDiv` and `$scroller` in the prototype?